pkg/controller/lease/controller.go

```diff
@@ -14,15 +14,14 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package nodelease
+package lease
```

Comment:
Putting this under pkg/controller seems confusing. The rest of the controllers under pkg/controller are control plane logic/controller being run by a controller-manager. This seems more like an implementation of heartbeat for a process (Kubelet) than a control plane controller.

Reply:
That's true. The lease controller is a heartbeat implementation. It is different from control plane controllers in terms of 1. it doesn't watch any resource 2. it doesn't run leader election-- each replica of a component (kubelet) runs the lease controller simultaneously. 

The other options I can think of:
1. [k8s.io/client-go/tools](https://github.com/kubernetes/client-go/tree/master/tools) - the behavior is similar to `leaderelection` (no watch; all replicas run simultaneously)
2. [pkg/util](https://github.com/kubernetes/kubernetes/tree/master/pkg/util) - we do have some [utility](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/filesystem/watcher.go#L73) that spawns goroutine, but it's less related to k8s resources. 
3. or we may create a new in-tree directory (e.g. `pkg/tools`, `pkg/client/tools`) to avoid making the existing ones confusing. 

@deads2k @caesarxuchao Any suggestions?

Reply:
We trust it for kubelets, so I think we have a good degree of confidence.  Because it is related to apiserver handling that would apply to any apiserver, I think it needs to be somewhere public.  I think your suggestion of client-go/tools is a good one.

Reply:
Thanks for the comments. I update the PR to put it under client-go/tools. Please take a look

Reply:
I disagree that client-go is the right repo. component-base was invented to have reusable functionality shared by our binaries. Hence, lease controller belongs there.

Reply:
Component-base hosts code and best practices used by [all the core components](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/wgs/0032-create-a-k8s-io-component-repo.md#history-and-motivation). This lease controller is not as critical IMO. 

On the other hand, the recently created [component-helpers](https://github.com/kubernetes/kubernetes/pull/92507) hosts helpers required by [two or more components](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/component-helpers#component-helpers), and the definition for a "helper" seems more arbitrary than a "base". I moved the controller to component-helpers. PTAL